streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -955,20 +969,28 @@ boolean maybeCommit() {
                     taskManager.activeTaskIds(), taskManager.standbyTaskIds(), now - lastCommitMs, commitTimeMs);
             }
 
-            committed = taskManager.commitAll();
+            committed = taskManager.commitAllActive();
             if (committed > 0) {
+                if (eosBetaEnabled) {
+                    producer.commitTransaction();
+                    producer.beginTransaction();
+                }
+            }
+            if (committed > 0 || taskManager.commitAllStandby() > 0) {
```

Comment:
prop: We could do
```
committed += taskManager.commitAllStandby();
if (committed > 0) 
```
for the rest of the logic so that we don't need a separate `if (committed > 0)` later

Reply:
We don't need to call `maybePurgeCommitedRecords()` if we only make progress on StandbyTasks -- if active tasks don't make progress, we know that there is nothing to be purged from repartition topics.

Reply:
That would be different from our existing logic as we care about combined results from active and standby:
```
committed = taskManager.commitAll();
            if (committed > 0) {
                final long intervalCommitLatency = advanceNowAndComputeLatency();
                commitSensor.record(intervalCommitLatency / (double) committed, now);

                // try to purge the committed records for repartition topics if possible
                taskManager.maybePurgeCommitedRecords();

                if (log.isDebugEnabled()) {
                    log.debug("Committed all active tasks {} and standby tasks {} in {}ms",
                        taskManager.activeTaskIds(), taskManager.standbyTaskIds(), intervalCommitLatency);
                }
            }
```
Do we have tests to cover this case change?

Reply:
Yes, it's different. @guozhangwang wrote this code about purging -- not sure if there is some specific reasoning for the existing code (I personally doubt it) -- the change did not break any tests hence, I assume we are fine (there is `PurgeRepartitionTopicIntegrationTest.java`)